### PR TITLE
Add Approval Workflow extension

### PR DESCRIPTION
**Pull request by jelz**

### CODE REVIEW

File: samples/basic-approval-workflow/src/app.js [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    if (typeof this.off === 'function') {
+      this.off();
+    }
+  }
+
+  render = () => {
+    const { log, users } = this.props;
+    const { currentVersion, busy } = this.state;
+
+    const last = log[0];
```
Comment (zcei) [inline samples/basic-approval-workflow/src/app.js]: As we're destructuring everywhere above:

```suggestion
    const [last] = log;
```

File: samples/basic-approval-workflow/src/app.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+          <Button
+            buttonType="negative"
+            onClick={this.cancelReviewRequest}
+            isFullWidth
+          >
+            Cancel review request
+          </Button>
+        </React.Fragment>
+      );
+    } if (isReviewer) {
```
Comment (zcei) [inline samples/basic-approval-workflow/src/app.js]: This should be either an `else if` or (even better) just on a newline 😆

File: samples/basic-approval-workflow/src/app.js [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+    const item = {
+      ...data,
+      eventType,
+      id: id(),
+      userId: sdk.user.sys.id,
+      version: currentVersion,
+    };
+
+    // Send a message for real-time sync and persistance
```
Comment (zcei) [inline samples/basic-approval-workflow/src/app.js]:
```suggestion
    // Send a message for real-time sync and persistence
```

File: samples/basic-approval-workflow/src/pubnub-client.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+  function addToLog({ entry, timetoken }) {
+    const validEntry = entry && typeof entry.id === 'string';
+    const validTimetoken = typeof timetoken === 'string';
+    const newEntry = !state.log.find(cur => cur.id === entry.id);
+
+
+    if (validEntry && validTimetoken && newEntry) {
+      // Timetoken is a 17-digit precision unix time (UTC)
+      const t = parseInt(timetoken, 10) / 10000;
```
Comment (zcei) [inline samples/basic-approval-workflow/src/pubnub-client.js]: `Number.MAX_SAFE_INTEGER` is lower than a 17 digit unix time (it has only 16 digits).
As we probably don't want to use `BigInt` here I'd just slice away the last four digits.

File: samples/basic-approval-workflow/src/pubnub-client.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    return false;
+  }
+
+  function getHistory() {
+    return new Promise((resolve, reject) => {
+      pubnub.history({
+        channel,
+        count: 25,
+        stringifiedTimeToken: true,
+      }, (status, res) => {
```
Comment (zcei) [inline samples/basic-approval-workflow/src/pubnub-client.js]: Why are you using the callback interface here? 
[There seems to be a Promise version as well](https://www.pubnub.com/docs/web-javascript/api-reference-storage-and-playback#history-basic-usage-using-promises)

File: samples/basic-approval-workflow/src/pubnub-client.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        } else {
+          resolve(res);
+        }
+      });
+    });
+  }
+
+  function publish(message) {
+    return new Promise((resolve, reject) => {
+      pubnub.publish({ channel, message }, (status, res) => {
```
Comment (zcei) [inline samples/basic-approval-workflow/src/pubnub-client.js]: Same, here, [there's a Promise version for publish as well](https://www.pubnub.com/docs/web-javascript/api-reference-publish-and-subscribe#publish-basic-usage-using-promises)

File: samples/basic-approval-workflow/src/pubnub-client.js (hunk 5 above)
Comment (jelz) [reply samples/basic-approval-workflow/src/pubnub-client.js]: Cool, wasn't aware of it!

**Pull request merged**